### ISSUE TITLE
Any API support python generator function?

### ISSUE THREAD

**Issue by Primus-zhao** [apache/incubator-mxnet#3215]
Hi, 
is there any API for python that supports getting data from generator? Specifically like the fit_generator API in keras: 
https://keras.io/models/sequential/

**Comment by mguillau** [comment]
If you look at the examples, like the one at [1], you see that mxnet already supports having generators in its fit() method and you can simply derive from the DataIter class [2] to make one in python.

Your class should provide the following methods: iter_next(), reset(), getdata(), getlabel(), getindex() and getpad(). As you write those methods, you should cast your data and labels into NDArray objects. For that you can use the mxnet.ndarray.array() function [3] to convert from numpy.

A notable difference with generators for Keras is that Keras expects an infinite continuous flow, while mxnet expects a StopIteration to be raised after each epoch, so make sure that iter_next() returns False when the epoch is over.

Hope this helps!

-- 
[1] https://github.com/dmlc/mxnet/blob/v0.7.0/example/image-classification/train_cifar10.py#L81
[2] https://github.com/dmlc/mxnet/blob/master/python/mxnet/io.py#L99
[3] https://github.com/dmlc/mxnet/blob/master/python/mxnet/ndarray.py#L919

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
